### ISSUE TITLE
Can not sync local repositories, always in a loading state

### ISSUE THREAD

**Issue by birdyson** [BloopAI/bloop#454]
MacOS Ventura 13.0 Apple Silicon M1
Version 0.3.4 (20230418.162902)

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/53157530/235974456-48d0797a-0a3c-47db-bf1c-533c44db44ed.png">

**Comment by birdyson** [comment]
I have resolved the issue. The root cause of the problem was my abnormal network status. After fixing it, I was able to synchronize the local repository code. However, as a user, I may not know whether the system is waiting due to my network issue or actually processing the request. Therefore, I suggest using a timeout mechanism or a progress bar to friendly notify the user about the current status of the processing request.
